core/src/snapshot_packager_service.rs

```diff
@@ -85,12 +91,16 @@ impl SnapshotPackagerService {
                             .push_snapshot_hash(snapshot_kind, (snapshot_slot, snapshot_hash));
                     }
 
+                    let snapshot_load_config = snapshot_mode.get_snapshot_load_config().clone();
+                    let incremental_snapshot_config =
+                        snapshot_load_config.incremental_snapshot_config.unwrap();
+
                     let (_, purge_archives_time_us) =
                         measure_us!(snapshot_utils::purge_old_snapshot_archives(
-                            &snapshot_config.full_snapshot_archives_dir,
-                            &snapshot_config.incremental_snapshot_archives_dir,
-                            snapshot_config.maximum_full_snapshot_archives_to_retain,
-                            snapshot_config.maximum_incremental_snapshot_archives_to_retain,
+                            &snapshot_load_config.full_snapshot_config.archives_dir,
+                            &incremental_snapshot_config.archives_dir, // FIXME: Should this param be optional too?
```

Comment:
probably you can split the function into two. One for full, and one for incremental snapshots. In a separate PR.